lib/view/widgets/extended_menu_fab.dart

```diff
@@ -0,0 +1,128 @@
+import 'package:flutter/material.dart';
+import 'package:superformula_test/core/resources/app_routes.dart';
+
+class ExtendedMenuFAB extends StatefulWidget {
+  const ExtendedMenuFAB({super.key});
+
+  @override
+  State<ExtendedMenuFAB> createState() => _ExtendedMenuFABState();
+}
+
+class _ExtendedMenuFABState extends State<ExtendedMenuFAB>
+    with SingleTickerProviderStateMixin {
+  late final AnimationController controller;
+  late final Animation<double> animation;
+  bool openedMenu = false;
+
+  @override
+  void initState() {
+    super.initState();
+    controller = AnimationController(
```

Comment:
**praise**: this is a great example of using `AnimationController`, very good 😄  I recommend you do use this in the previous case as well